toolchain/check/import_cpp.cpp

```diff
@@ -70,13 +68,81 @@ auto ImportCppFiles(
         IntAsSelect, IntAsSelect, IntAsSelect, std::string);
     context.emitter().Emit(loc, CppInteropParseError, num_errors, num_warnings,
                            num_imports, diagnostics_str);
-  } else if (num_warnings > 0) {
+    return std::nullopt;
+  }
+  if (num_warnings > 0) {
     CARBON_DIAGNOSTIC(CppInteropParseWarning, Warning,
                       "{0} warning{0:s} in `Cpp` {1} import{1:s}:\n{2}",
                       IntAsSelect, IntAsSelect, std::string);
     context.emitter().Emit(loc, CppInteropParseWarning, num_warnings,
                            num_imports, diagnostics_str);
   }
+  return ast;
+}
+
+static auto AddNamespace(Context& context, IdentifierId cpp_package_id,
+                         SemIR::InstId first_import_decl_id) -> void {
+  // TODO: Extract and deudplicate common logic with import.cpp.
+  auto name_id = SemIR::NameId::ForIdentifier(cpp_package_id);
+  SemIR::TypeId namespace_type_id =
+      context.GetSingletonType(SemIR::NamespaceType::SingletonInstId);
+
+  auto* parent_scope = &context.name_scopes().Get(SemIR::NameScopeId::Package);
+  auto [inserted, entry_id] = parent_scope->LookupOrAdd(
+      name_id,
+      // This InstId is temporary and would be overridden if used.
+      first_import_decl_id, SemIR::AccessKind::Public);
+  CARBON_CHECK(inserted);
+
+  CARBON_CHECK(first_import_decl_id.has_value());
```

Comment:
GetLocId() checks the same thing (via `>= 0`) so I think this isn't needed here.

Reply:
Done.